From: Hector Palacios <hector.palacios@digi.com>
Date: Tue, 19 Apr 2011 11:55:31 +0200
Subject: [PATCH] ncb: fixed transposed parameters in memset

Doesn't seem to be critical, but we'd better have the buffer with zeros
than with data.

Signed-off-by: Hector Palacios <hector.palacios@digi.com>
---
 src/ncb.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ncb.c b/src/ncb.c
index 927ca01..6e6f21d 100644
--- a/src/ncb.c
+++ b/src/ncb.c
@@ -332,7 +332,7 @@ int ncb_encrypt(BootBlockStruct_t *ncb, void *target, size_t size, int version)
 	assert(version == 0 || version == 1 || version == 3);
 	assert(size >= sizeof(BootBlockStruct_t));
 
-	memset(target, size, 0);
+	memset(target, 0, size);
 
 	switch (version)
 	{
@@ -377,7 +377,7 @@ int fcb_encrypt(V1_ROM_BootBlockStruct_t *fcb, void *target, size_t size, int ve
 	// Clear out the target.
 	//----------------------------------------------------------------------
 
-	memset(target, size, 0);
+	memset(target, 0, size);
 
 	//----------------------------------------------------------------------
 	// Compute the checksum.
